### PR TITLE
feat: impliment Bresenham's algorithm in drawLine

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**

### CODE REVIEW

**Comment by targos** [comment]
Please install and use https://github.com/w8r/bresenham-zingl (don't copy the code)

**Comment by wadjih-bencheikh18** [comment]
> Please install and use https://github.com/w8r/bresenham-zingl (don't copy the code)

bresenham-zingl does not have ts version, it may cause issues

**Comment by wadjih-bencheikh18** [comment]
Do i have to create new drawline using Bresenham's algorithm or replace the function that already exist?

**Comment by targos** [comment]
The goal is to replace the implementation, in other words call the library from `drawLineOnIjs` and `drawLineOnMask`

**Comment by targos** [comment]
> > Please install and use https://github.com/w8r/bresenham-zingl (don't copy the code)
> 
> bresenham-zingl does not have ts version, it may cause issues

You can create a file in `src/types/bresenham-zingl.d.ts` and define the types that we need.

**Comment by wadjih-bencheikh18** [comment]
> The goal is to replace the implementation, in other words call the library from `drawLineOnIjs` and `drawLineOnMask`

i created `drawLineOnIjs` using `bresenham-zingl`

File: package.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
     "@types/robust-point-in-polygon": "^1.0.2",
     "@vitejs/plugin-react": "^1.3.2",
     "autoprefixer": "^10.4.7",
+    "bresenham-zingl": "^0.1.1",
```
Comment (targos) [inline package.json:61]: this is not a development dependency.

File: src/draw/drawLineOnIjs.ts [hunk 2]
```diff
@@ -42,37 +44,39 @@ export function drawLineOnIjs(
...
-      } else {
-        err += dColumn;
-        row += rowIncrement;
-      }
-    }
+  const numberChannels = Math.min(newImage.channels, color.length);
+  if (
+    [ImageColorModel.GREYA, ImageColorModel.RGBA].includes(image.colorModel)
+  ) {
+    lineAA(
```
Comment (targos) [inline src/draw/drawLineOnIjs.ts:52]: Anti-aliasing will be an option in the fututre but we don't want it for now

Comment (wadjih-bencheikh18) [reply src/draw/drawLineOnIjs.ts:52]: ok removed

File: package.json (hunk 1 above)
Comment (wadjih-bencheikh18) [reply package.json]: added to dependencies

**Comment by targos** [comment]
Lgtm. Can you check the CI errors?

**Comment by targos** [comment]
OK I know what the problem is. Working on it.

**Pull request merged**